--- origsrc/kdelibs-3.5.9/kio/kio/global.cpp	2006-05-22 13:14:17.000000000 -0500
+++ src/kdelibs-3.5.9/kio/kio/global.cpp	2008-02-20 08:54:09.437500000 -0600
@@ -1347,7 +1347,11 @@
 #define GETMNTENT(file, var) ((var = getmntent(file)) != 0)
 #define MOUNTPOINT(var) var->mnt_dir
 #define MOUNTTYPE(var) var->mnt_type
+#if defined(__CYGWIN__)
+#define HASMNTOPT(var, opt) 0
+#else
 #define HASMNTOPT(var, opt) hasmntopt(var, opt)
+#endif
 #define FSNAME(var) var->mnt_fsname
 #elif defined(_AIX)
 /* we don't need this stuff */
