--- origsrc/kdelibs-3.5.9/kio/kssl/kopenssl.cc	2006-07-22 03:16:39.000000000 -0500
+++ src/kdelibs-3.5.9/kio/kssl/kopenssl.cc	2008-02-20 08:54:09.531250000 -0600
@@ -289,18 +289,14 @@
 #elif defined(__CYGWIN__)
    libpaths << "/usr/bin/"             
    		<< "/usr/local/bin"             
-   		<< "/usr/local/openssl/bin"     
-   		<< "/opt/openssl/bin"           
-   		<< "/opt/kde3/bin"              
    		<< "";                          
                                        
-   libnamess << "cygssl-0.9.7.dll"     
-		 << "cygssl.dll"                    
-		 << "libssl.dll"                    
+   libnamess << "cygssl-0.9.8.dll"
+		 << "cygssl-0.9.7.dll"     
 		 << "";                         
                                        
-   libnamesc << "cygcrypto.dll"        
-		 << "libcrypto.dll"                 
+   libnamesc << "cygcrypto-0.9.8.dll"
+		 << "cygcrypto-0.9.7.dll"
 		 << "";                         
 #else
    libpaths
