--- origsrc/kdelibs-3.5.9/kdecore/kextsock.cpp	2007-10-08 04:51:55.000000000 -0500
+++ src/kdelibs-3.5.9/kdecore/kextsock.cpp	2008-02-20 08:54:09.000000000 -0600
@@ -44,6 +44,7 @@
 
 #include "kresolver.h"
 
+#include "netsupp.h"
 #include "kdebug.h"
 #include "kextsock.h"
 #include "ksockaddr.h"
--- origsrc/kdelibs-3.5.9/kdecore/netsupp.h	2005-10-10 10:06:03.000000000 -0500
+++ src/kdelibs-3.5.9/kdecore/netsupp.h	2008-02-20 08:54:09.015625000 -0600
@@ -29,7 +29,7 @@
 
 #include <sys/socket.h>
 #include <netdb.h>
-//#include "ksockaddr.h"
+#include "ksocks.h"
 
 #ifdef __CYGWIN__                 
 typedef unsigned ksocklen_t;
--- origsrc/kdelibs-3.5.9/kdecore/network/kresolver.cpp	2006-07-22 03:16:40.000000000 -0500
+++ src/kdelibs-3.5.9/kdecore/network/kresolver.cpp	2008-02-20 08:54:09.015625000 -0600
@@ -1097,7 +1097,7 @@
   QCString retval;
   char buf[65];
 
-  Q_UINT32* ucs4 = new Q_UINT32[label.length() + 1];
+  uint32_t* ucs4 = new uint32_t[label.length() + 1];
 
   uint i;
   for (i = 0; i < label.length(); i++)
@@ -1121,15 +1121,15 @@
   // We have idna.h, so we can use the idna_to_unicode
   // function :)
 
-  Q_UINT32 *ucs4_input, *ucs4_output;
+  uint32_t *ucs4_input, *ucs4_output;
   size_t outlen;
 
-  ucs4_input = new Q_UINT32[label.length() + 1];
+  ucs4_input = new uint32_t[label.length() + 1];
   for (uint i = 0; i < label.length(); i++)
     ucs4_input[i] = (unsigned long)label[i].unicode();
 
   // try the same length for output
-  ucs4_output = new Q_UINT32[outlen = label.length()];
+  ucs4_output = new uint32_t[outlen = label.length()];
 
   idna_to_unicode_44i(ucs4_input, label.length(),
 		      ucs4_output, &outlen,
@@ -1139,7 +1139,7 @@
     {
       // it must have failed
       delete [] ucs4_output;
-      ucs4_output = new Q_UINT32[outlen];
+      ucs4_output = new uint32_t[outlen];
 
       idna_to_unicode_44i(ucs4_input, label.length(),
 			  ucs4_output, &outlen,
--- origsrc/kdelibs-3.5.9/kwallet/backend/sha1.cc	2006-05-22 13:14:18.000000000 -0500
+++ src/kdelibs-3.5.9/kwallet/backend/sha1.cc	2008-02-20 08:54:10.203125000 -0600
@@ -32,6 +32,9 @@
 #ifdef HAVE_STDINT_H
 #include <stdint.h>
 #endif
+#ifdef HAVE_INTTYPES_H
+#include <inttypes.h>
+#endif
 
 #include "sha1.h"
 #include <string.h>
