--- origsrc/kdelibs-4.0.99/kio/kssl/kopenssl.cpp	2008-07-03 00:06:28.000000000 -0500
+++ src/kdelibs-4.0.99/kio/kssl/kopenssl.cpp	2008-07-21 00:15:53.250000000 -0500
@@ -315,19 +315,14 @@
    }
 #elif defined(__CYGWIN__)
    libpaths << "/usr/bin/"
-   		<< "/usr/local/bin"
-   		<< "/usr/local/openssl/bin"
-   		<< "/opt/openssl/bin"
-   		<< "/opt/kde3/bin"
    		<< "";
 
-   libnamess << "cygssl-0.9.7.dll"
-		 << "cygssl.dll"
-		 << "libssl.dll"
+   libnamess << "cygssl-0.9.8.dll"
+		 << "cygssl-0.9.7.dll"
 		 << "";
 
-   libnamesc << "cygcrypto.dll"
-		 << "libcrypto.dll"
+   libnamesc << "cygcrypto-0.9.8.dll"
+		 << "cygcrypto-0.9.7.dll"
 		 << "";
 #else
    libpaths
