--- origsrc/kdelibs-4.0.99/kdecore/network/k3socks.h	2008-05-21 06:09:04.000000000 -0500
+++ src/kdelibs-4.0.99/kdecore/network/k3socks.h	2008-07-21 00:11:41.390625000 -0500
@@ -34,9 +34,6 @@
 class KConfigGroup;
 struct sockaddr;
 
-#ifdef __CYGWIN__
-typedef unsigned ksocklen_t;
-#endif
 
 /**
  * This class provides you with an interface to a 
--- origsrc/kdelibs-4.0.99/kdecore/network/k3sockssocketdevice.cpp	2008-05-21 06:09:04.000000000 -0500
+++ src/kdelibs-4.0.99/kdecore/network/k3sockssocketdevice.cpp	2008-07-21 00:11:54.062500000 -0500
@@ -30,10 +30,6 @@
 #include <unistd.h>
 #endif
 
-#ifdef __CYGWIN__
-#undef kde_socklen_t
-#define kde_socklen_t ksocklen_t
-#endif
 
 #include <QCoreApplication>
 #ifndef KDE_USE_FINAL
--- origsrc/kdelibs-4.0.99/kdecore/network/netsupp.h	2008-05-21 06:09:04.000000000 -0500
+++ src/kdelibs-4.0.99/kdecore/network/netsupp.h	2008-07-21 00:11:47.875000000 -0500
@@ -30,9 +30,6 @@
 #include <netdb.h>
 //#include "ksockaddr.h"
 
-#ifdef __CYGWIN__
-typedef unsigned ksocklen_t;
-#endif
 
 /*
  * Seems some systems don't know about AF_LOCAL
