--- origsrc/kdelibs-4.0.99/kdecore/tests/klibloadertest.cpp	2008-05-21 06:09:07.000000000 -0500
+++ src/kdelibs-4.0.99/kdecore/tests/klibloadertest.cpp	2008-07-21 20:44:36.015625000 -0500
@@ -53,7 +53,7 @@
     const QString library = KLibLoader::findLibrary( s_kgenericFactoryModule );
     QVERIFY( !library.isEmpty() );
     const QString libraryPath = QFileInfo( library ).canonicalFilePath();
-#ifdef Q_OS_WIN
+#if defined(Q_OS_WIN) || defined(Q_OS_CYGWIN)
     const QString expectedPath = QFileInfo( QDir::currentPath() + "/../../lib/" + s_kgenericFactoryModule + ".dll" ).canonicalFilePath();
 #else
     const QString expectedPath = QFileInfo( QDir::currentPath() + "/../../lib/" + s_kgenericFactoryModule + ".so" ).canonicalFilePath();
--- origsrc/kdelibs-4.0.99/kdecore/util/kpluginloader.cpp	2008-05-21 06:09:15.000000000 -0500
+++ src/kdelibs-4.0.99/kdecore/util/kpluginloader.cpp	2008-07-21 20:57:10.296875000 -0500
@@ -54,7 +54,7 @@
 
 inline QString makeLibName( const QString &libname )
 {
-#ifdef Q_OS_WIN
+#if defined(Q_OS_WIN) || defined(Q_OS_CYGWIN)
     if (!libname.endsWith(".dll"))
         return libname + ".dll";
     return libname;
