--- origsrc/kdelibs-4.1.4/kdemacros.h.cmake	2009-01-06 17:38:25.000000000 -0600
+++ src/kdelibs-4.1.4/kdemacros.h.cmake	2009-01-14 21:20:42.827750000 -0600
@@ -73,7 +73,7 @@
 #define KDE_NO_EXPORT __attribute__ ((visibility("hidden")))
 #define KDE_EXPORT __attribute__ ((visibility("default")))
 #define KDE_IMPORT __attribute__ ((visibility("default")))
-#elif defined(_WIN32) || defined(_WIN64)
+#elif defined(_WIN32) || defined(_WIN64) || defined(__CYGWIN__)
 #define KDE_NO_EXPORT
 #define KDE_EXPORT __declspec(dllexport)
 #define KDE_IMPORT __declspec(dllimport)
--- origsrc/kdelibs-4.1.4/kpty/kpty_export.h	2009-01-06 17:35:57.000000000 -0600
+++ src/kdelibs-4.1.4/kpty/kpty_export.h	2009-01-15 17:33:47.447813900 -0600
@@ -24,7 +24,7 @@
 #include <kdemacros.h>
 
 #ifndef KPTY_EXPORT
-# if defined(MAKE_KDECORE_LIB)
+# if defined(MAKE_KPTY_LIB)
    /* We are building this library */ 
 #  define KPTY_EXPORT KDE_EXPORT
 # else
