--- origsrc/kdelibs-4.1.4/kio/kio/slaveinterface_p.h	2009-01-06 17:35:41.000000000 -0600
+++ src/kdelibs-4.1.4/kio/kio/slaveinterface_p.h	2009-01-15 21:08:53.338438900 -0600
@@ -21,6 +21,7 @@
 
 #include "global.h"
 #include "connection.h"
+#include <sys/time.h>
 #include <QTimer>
 
 static const unsigned int max_nums = 8;
--- origsrc/kdelibs-4.1.4/kpty/kpty.cpp	2009-01-06 17:35:57.000000000 -0600
+++ src/kdelibs-4.1.4/kpty/kpty.cpp	2009-01-15 17:28:14.447813900 -0600
@@ -483,7 +483,7 @@ void KPty::login(const char *user, const
     addToUtmp(d->ttyName, remotehost, d->masterFd);
     Q_UNUSED(user);
 #else
-# ifdef HAVE_UTMPX
+# ifdef HAVE_LOGINX
     struct utmpx l_struct;
 # else
     struct utmp l_struct;
@@ -514,7 +514,7 @@ void KPty::login(const char *user, const
 #  endif
 # endif
 
-# ifdef HAVE_UTMPX
+# ifdef HAVE_LOGINX
     gettimeofday(&l_struct.ut_tv, 0);
 # else
     l_struct.ut_time = time(0);
@@ -562,7 +562,7 @@ void KPty::logout()
 #else
     Q_D(KPty);
 
-    const char *str_ptr = d->ttyName.data();
+    char *str_ptr = d->ttyName.data();
     if (!memcmp(str_ptr, "/dev/", 5))
         str_ptr += 5;
 # ifdef __GLIBC__
--- origsrc/kdelibs-4.1.4/kpty/kptydevice.cpp	2009-01-06 17:35:57.000000000 -0600
+++ src/kdelibs-4.1.4/kpty/kptydevice.cpp	2009-01-15 17:29:32.510313900 -0600
@@ -42,6 +42,10 @@
 # include <sys/time.h>
 #endif
 
+#ifdef __CYGWIN__
+#define FIONREAD TIOCINQ
+#endif
+
 #define KMAXINT ((int)(~0U >> 1))
 
 /////////////////////////////////////////////////////
--- origsrc/kdelibs-4.1.4/nepomuk/core/resourcefiltermodel.cpp	2009-01-06 17:35:20.000000000 -0600
+++ src/kdelibs-4.1.4/nepomuk/core/resourcefiltermodel.cpp	2009-01-15 19:54:38.713438900 -0600
@@ -47,11 +47,13 @@ uint Soprano::qHash( const Soprano::Node
     return qHash( node.toString() );
 }
 
+#if 0
 // @deprecated: just for keeping binary compatibility
 uint qHash( const Soprano::Node& node )
 {
     return Soprano::qHash( node );
 }
+#endif
 
 class Nepomuk::ResourceFilterModel::Private
 {
