--- origsrc/kdelibs-4.2.4/solid/solid/CMakeLists.txt	2008-11-19 04:20:13.000000000 -0600
+++ src/kdelibs-4.2.4/solid/solid/CMakeLists.txt	2009-07-02 18:09:44.183981200 -0500
@@ -102,7 +102,7 @@ set(solid_LIB_SRCS 
    backends/fakehw/fakevolume.cpp
 )
 
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32 AND NOT APPLE AND NOT CYGWIN)
    message(STATUS "-- Building Solid HAL backend." )
    set(solid_LIB_SRCS ${solid_LIB_SRCS}
 
@@ -129,7 +129,7 @@ if(NOT WIN32 AND NOT APPLE)
    backends/hal/halvolume.cpp
    
    )
-endif(NOT WIN32 AND NOT APPLE)
+endif(NOT WIN32 AND NOT APPLE AND NOT CYGWIN)
 
 # TODO: mingw
 if(MSVC)
--- origsrc/kdelibs-4.2.4/solid/solid/managerbase.cpp	2008-05-29 06:20:07.000000000 -0500
+++ src/kdelibs-4.2.4/solid/solid/managerbase.cpp	2009-07-02 18:13:55.496571200 -0500
@@ -23,7 +23,7 @@
 
 #include "backends/fakehw/fakemanager.h"
 
-#if defined (Q_OS_MAC)
+#if defined (Q_OS_MAC) || defined (Q_OS_CYGWIN)
 #elif defined (Q_OS_UNIX)
 #include "backends/hal/halmanager.h"
 #elif defined (_MSC_VER) // TODO: mingw
@@ -47,7 +47,7 @@ void Solid::ManagerBasePrivate::loadBack
     if (!solidFakeXml.isEmpty()) {
         m_backend = new Solid::Backends::Fake::FakeManager(0, solidFakeXml);
     } else {
-        #if defined (Q_OS_MAC)
+        #if defined (Q_OS_MAC) || defined (Q_OS_CYGWIN)
         #elif defined (Q_OS_UNIX)
             m_backend = new Solid::Backends::Hal::HalManager(0);
         #elif defined (_MSC_VER) // TODO: mingw
