--- origsrc/kdelibs-4.2.4/kdecore/localization/klocale.cpp	2009-05-28 14:22:22.000000000 -0500
+++ src/kdelibs-4.2.4/kdecore/localization/klocale.cpp	2009-06-28 17:33:35.660393000 -0500
@@ -2092,7 +2092,7 @@ void KLocalePrivate::initEncoding()
   codecForEncoding = 0;
 
   // This all made more sense when we still had the EncodingEnum config key.
-#if defined(HAVE_LANGINFO_H) && !defined(Q_OS_WIN)
+#if defined(HAVE_LANGINFO_H) && !defined(Q_OS_WIN) && !defined(Q_OS_CYGWIN)
   // Qt since 4.2 always returns 'System' as codecForLocale and KDE (for example KEncodingFileDialog)
   // expects real encoding name. So on systems that have langinfo.h use nl_langinfo instead,
   // just like Qt compiled without iconv does. Windows already has its own workaround
