--- origsrc/kdelibs-4.6.4/kjsembed/kjsembed/CMakeLists.txt	2011-04-01 08:55:51.000000000 -0500
+++ src/kdelibs-4.6.4/kjsembed/kjsembed/CMakeLists.txt	2011-06-12 13:40:25.677304200 -0500
@@ -68,6 +68,7 @@ kde4_add_library(${KJSEMBEDLIBNAME} ${LI
 
 target_link_libraries(${KJSEMBEDLIBNAME} ${KDE4_KDECORE_LIBS} ${QT_QTUITOOLS_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTSVG_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTCORE_LIBRARY} ${KJSLIBNAME} )
 
+set_target_properties(${KJSEMBEDLIBNAME} PROPERTIES LINK_FLAGS "-Wl,--exclude-libs -Wl,libQtUiTools.a")
 set_target_properties(${KJSEMBEDLIBNAME} PROPERTIES VERSION ${GENERIC_LIB_VERSION} SOVERSION ${GENERIC_LIB_SOVERSION} )
 install(TARGETS ${KJSEMBEDLIBNAME} EXPORT kdelibsLibraryTargets ${INSTALL_TARGETS_DEFAULT_ARGS})
 
--- origsrc/kdelibs-4.6.4/plasma/CMakeLists.txt	2011-04-01 08:56:18.000000000 -0500
+++ src/kdelibs-4.6.4/plasma/CMakeLists.txt	2011-06-12 13:39:35.554416800 -0500
@@ -252,6 +252,8 @@ target_link_libraries(plasma ${QT_QTUITO
                              ${QT_QTSCRIPT_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTSQL_LIBRARY} ${QT_QTDECLARATIVE_LIBRARY}
                              ${KDE4_KDEUI_LIBS} kdnssd threadweaver ${PLASMA_EXTRA_LIBS})
 
+set_target_properties(plasma PROPERTIES LINK_FLAGS "-Wl,--exclude-libs -Wl,libQtUiTools.a")
+
 if(QCA2_FOUND)
    target_link_libraries(plasma ${QCA2_LIBRARIES})
 endif(QCA2_FOUND)

