--- origsrc/kdelibs-4.6.4/kdecore/util/kshareddatacache_p.h	2011-04-01 08:56:18.000000000 -0500
+++ src/kdelibs-4.6.4/kdecore/util/kshareddatacache_p.h	2011-06-12 17:58:27.220795800 -0500
@@ -311,7 +311,9 @@ static SharedLockId findBestSharedLock()
         sem_t tempSemaphore;
         QSharedPointer<KSDCLock> tempLock(0);
         if (timeoutsSupported) {
+#ifdef KSDC_TIMEOUTS_SUPPORTED
             tempLock = QSharedPointer<KSDCLock>(new semaphoreTimedLock(tempSemaphore));
+#endif
         }
         else {
             tempLock = QSharedPointer<KSDCLock>(new semaphoreLock(tempSemaphore));
--- origsrc/kdelibs-4.6.4/kdeui/kernel/kuniqueapplication.cpp	2011-04-01 08:55:50.000000000 -0500
+++ src/kdelibs-4.6.4/kdeui/kernel/kuniqueapplication.cpp	2011-06-12 13:30:58.603917700 -0500
@@ -117,7 +117,7 @@ KUniqueApplication::start(StartFlags fla
   s_kuniqueapplication_startCalled = true;
 
   addCmdLineOptions(); // Make sure to add cmd line options
-#if defined(Q_WS_WIN) || defined(Q_WS_MACX)
+#if defined(Q_WS_WIN) || defined(Q_WS_MACX) || defined(Q_OS_CYGWIN)
   Private::s_nofork = true;
 #else
   KCmdLineArgs *args = KCmdLineArgs::parsedArgs("kuniqueapp");
--- origsrc/kdelibs-4.6.4/kio/kio/slaveinterface_p.h	2011-04-01 08:55:51.000000000 -0500
+++ src/kdelibs-4.6.4/kio/kio/slaveinterface_p.h	2011-06-12 20:02:23.034393300 -0500
@@ -21,6 +21,7 @@
 
 #include "global.h"
 #include "connection.h"
+#include <sys/time.h>
 #include <QtCore/QTimer>
 #include <QtNetwork/QHostInfo>
 
--- origsrc/kdelibs-4.6.4/kpty/kpty.cpp	2011-04-01 08:55:51.000000000 -0500
+++ src/kdelibs-4.6.4/kpty/kpty.cpp	2011-06-12 13:17:04.737664500 -0500
@@ -506,7 +506,7 @@ void KPty::login(const char *user, const
     addToUtmp(d->ttyName, remotehost, d->masterFd);
     Q_UNUSED(user);
 #else
-# ifdef HAVE_UTMPX
+# ifdef HAVE_LOGINX
     struct utmpx l_struct;
 # else
     struct utmp l_struct;
@@ -537,7 +537,7 @@ void KPty::login(const char *user, const
 #  endif
 # endif
 
-# ifdef HAVE_UTMPX
+# ifdef HAVE_LOGINX
     gettimeofday(&l_struct.ut_tv, 0);
 # else
     l_struct.ut_time = time(0);

