--- origsrc/kdelibs-4.6.4/kded/CMakeLists.txt	2011-04-01 08:55:50.000000000 -0500
+++ src/kdelibs-4.6.4/kded/CMakeLists.txt	2011-06-12 13:34:35.539894400 -0500
@@ -74,7 +74,7 @@ install(TARGETS kbuildsycoca4 ${INSTALL_
 if (WIN32)
 install( FILES applications.menu  DESTINATION  ${SHARE_INSTALL_PREFIX}/xdg/menus )
 else (WIN32)
-install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus )
+install( FILES applications.menu  DESTINATION  ${SYSCONF_INSTALL_DIR}/xdg/menus RENAME kde4-applications.menu )
 endif (WIN32)
 install( FILES kdedmodule.desktop DESTINATION  ${SERVICETYPES_INSTALL_DIR} )
 install( FILES kded.upd           DESTINATION  ${DATA_INSTALL_DIR}/kconf_update )
--- origsrc/kdelibs-4.6.4/kded/vfolder_menu.cpp	2011-04-01 08:55:50.000000000 -0500
+++ src/kdelibs-4.6.4/kded/vfolder_menu.cpp	2011-06-12 13:35:56.287234900 -0500
@@ -770,8 +770,9 @@ VFolderMenu::locateMenuFile(const QStrin
    QString result;
 
    QString xdgMenuPrefix = QString::fromLocal8Bit(qgetenv("XDG_MENU_PREFIX"));
-   if (!xdgMenuPrefix.isEmpty())
-   {
+   if (xdgMenuPrefix.isEmpty())
+      xdgMenuPrefix = QString("kde4-");
+
       QFileInfo fileInfo(fileName);
 
       QString fileNameOnly = fileInfo.fileName();
@@ -781,7 +782,6 @@ VFolderMenu::locateMenuFile(const QStrin
       QString baseName = QDir::cleanPath(m_docInfo.baseDir +
                                          fileInfo.path() + '/' + fileNameOnly);
       result = KStandardDirs::locate("xdgconf-menu", baseName);
-   }
 
    if (result.isEmpty())
    {

